store/src/speedtest.rs

```diff
@@ -48,3 +57,55 @@ impl TryFrom<SpeedtestReqV1> for CellSpeedtest {
         })
     }
 }
+
+impl TryFrom<SpeedtestIngestReportV1> for CellSpeedtestIngestReport {
+    type Error = Error;
+    fn try_from(v: SpeedtestIngestReportV1) -> Result<Self> {
+        Ok(Self {
+            received_timestamp: Utc.timestamp_millis(v.received_timestamp as i64),
+            report: TryFrom::try_from(v.report.unwrap())?,
+        })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use hex_literal::hex;
+    use prost::Message;
+
+    const PK_BYTES: [u8; 33] =
+        hex!("008f23e96ab6bbff48c8923cac831dc97111bcf33dba9f5a8539c00f9d93551af1");
+
+    #[test]
+    fn decode_proto_speed_test_ingest_report_to_internal_struct() {
+        let now = Utc::now().timestamp_millis();
+        let report = SpeedtestIngestReportV1 {
+            received_timestamp: now as u64,
+            report: Some(SpeedtestReqV1 {
+                pub_key: PK_BYTES.to_vec(),
+                serial: "serial".to_string(),
+                timestamp: now as u64,
+                upload_speed: 6,
+                download_speed: 2,
+                latency: 1,
+                signature: vec![],
+            }),
+        };
+
+        let mut buffer = vec![];
+
+        report
+            .encode(&mut buffer)
```

Comment:
use `encode_to_vec`

Reply:
Fixed 24ac815, can't believe I missed that function